builder/static_code.go

```diff
@@ -846,13 +878,15 @@ func (p *parser) parseChoiceExpr(ch *choiceExpr) (interface{}, bool) {
 	}
 
 	// {{ end }} ==template==
+	state := p.cloneState()
```

Comment:
I agree, if you have `state := p.cloneState()` outside the for-loop, then you need the copying in `restoreState()`. My solution was to place `cloneState` inside the following for-loop.

Reply:
ok. i revert it to your solution. And i add comment with restricts. Maybe we should use run-time checks for restrictions? Or this is useless increase of complexity?